core/src/core-plugins.ts

```diff
@@ -318,7 +325,15 @@ export const buildRequestInit = (
 
   // If body is already a string, then pass it through as-is.
   if (typeof options.data === 'string') {
-    output.body = options.data;
+    if (options?.gzipCompression && options.headers) {
+      options.headers['Content-Encoding'] = 'gzip';
+      output.headers = options.headers;
+
+      const gzippedData: Uint8Array = Pako.gzip(options.data);
```

Comment:
I'm not a fan of adding this dependency to core. Is there a reason you can't just gzip and gunzip on the client side from javascript on your side?

Reply:
the thing is that we could do that, but why we switched to capacitor http is, that we need background functionality of the http services and therefore we started using capacitor-community/http which brought us here. The Pako compression could be done client side in the app code, but we moved it to the web part of the plugin to complete the gzip functionality for all platforms. I didn't find any good code example to do it without Pako, maybe this dependency can be kept?